civis/service_client.py

```diff
@@ -2,41 +2,37 @@
 
 from collections import OrderedDict
 import re
-import warnings
 
 from jsonref import JsonRef
 import requests
 import six
 
 from civis import APIClient
-from civis.base import Endpoint, CivisAPIError, tostr_urljoin
+from civis.base import Endpoint, tostr_urljoin
 from civis.resources._resources import parse_method
 from civis._utils import to_camelcase
 
 
-def _get_service(service_id):
-    try:
-        client = APIClient()
-        service = client.services.get(service_id)
-    except CivisAPIError as err:
-        msg = ('There was an issue '
-               'finding service with ID {}.').format(service_id)
-        six.raise_from(ValueError(msg), err)
+def _get_service(client):
+    if client._api_key:
```

Comment:
I don't think having the api key as an argument is a wise interface to have. When using `ServiceClient`, a novice user could e.g. accidentally commit their API key to a public repo.

Reply:
@patr1ckm, thats a good point. I brought this up to @elsander before adding it, and they thought we should include it so we could support different work styles like APIClient does. Your concern is valid though, but I see ServiceClient as being an internal tool, with novice users mainly using the clients generated from it instead, like we plan to do with ToolboxClient(). Thoughts?

Reply:
Ah, thanks for the additional clarity!